presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -230,6 +234,41 @@ private static void generateConstructor(ClassDefinition classDefinition,
                     .putField(sizeField);
         }
 
+        constructor.comment("Set channelArrays field");
+
+        constructor.append(thisVariable.setField(
+                channelArraysField,
+                newInstance(ArrayList.class)));
+
+        if (!channelFields.isEmpty()) {
```

Comment:
this has about the same number of lines as `PageProcessorCompiler.fieldReferenceCompiler`. Compiling filter here directly would make this changes unnecessary.